### PR TITLE
[WIP] APP-214 Enable 'docker app validate' for experimental mode only

### PR DESCRIPTION
**Pull request by aiordache**
Signed-off-by: Anca Iordache <anca.iordache@docker.com>

Fixes  [APP-214](https://docker.atlassian.net/browse/APP-214)

Requires [PR-2095](https://github.com/docker/cli/pull/2095)

**- What I did**
Enabled validate command only for experimental mode

**- How I did it**
Added validate command to the list of commands only if experimental flag is on. 
**- How to verify it**
```
$ DOCKER_CLI_EXPERIMENTAL="disabled" docker app --help
Usage:	docker app COMMAND
Commands:
  ...
  run         Run an App from an App image
  update      Update a running App
```
```
$ DOCKER_CLI_EXPERIMENTAL="enabled" docker app --help
Usage:	docker app COMMAND
Commands:
  ...
  run         Run an App from an App image
  update      Update a running App
  validate    Check that an App definition (.dockerapp) is syntactically correct
```
```
$ DOCKER_CLI_EXPERIMENTAL="enabled" docker app validate
Validated "/home/anca/go/src/github.com/docker/app/examples/test.dockerapp"
```
```
$ DOCKER_CLI_EXPERIMENTAL="disabled" docker app validate
"validate" is not a docker app command
See 'docker app --help'
```
**- Description for the changelog**
Add validate to the list of docker app commands only in experimental mode.

**- A picture of a cute animal (not mandatory but encouraged)**
8d1

### CODE REVIEW

**Comment by thaJeztah** [comment]
looks like there's a merge-conflict, so might need a rebase.

File: internal/commands/root.go [hunk 1]
```diff
@@ -67,11 +68,23 @@ func addCommands(cmd *cobra.Command, dockerCli command.Cli) {
...
 		bundleCmd(dockerCli),
 		pushCmd(dockerCli),
 		pullCmd(dockerCli),
-	)
+		validateCmd(),
+	}
+
+	isExperimentalMode := dockerCli.ClientInfo().HasExperimental
+	for _, ccmd := range listOfCommands {
+		value, ok := ccmd.Annotations["experimental"]
```
Comment (silvin-lubecki) [inline internal/commands/root.go]: What do you think of:
```go
switch ccmd.Annotations["experimental"]{
case "true":
    if isExperimentalMode {
        cmd.AddCommand(ccmd)
    }
default:
    cmd.AddCommand(ccmd)
}
```

**Comment by silvin-lubecki** [comment]
The tests are failing.

File: internal/commands/root.go (hunk 1 above)
Comment (aiordache) [reply internal/commands/root.go]: Thanks! Updated it.

**Comment by ndeloof** [comment]
What's the status of this one ? Shall we better close ?

**Comment by aiordache** [comment]
I am not sure about this :(

I'll rebase the PR on cli and wait for it to be accepted and merged, we are
waiting for some feedback from Sebastiaan on it. Then, probably we can
update the vendor in the PR for docker app....?

File: Gopkg.toml [hunk 2]
```diff
@@ -113,6 +113,11 @@ required = ["github.com/wadey/gocovmerge"]
   name = "github.com/opencontainers/runtime-spec"
   revision = "29686dbc5559d93fb1ef402eeda3e35c38d75af4"
 
+[[override]]
+  name = "github.com/Microsoft/hcsshim"
+  revision = "2226e083fc390003ae5aa8325c3c92789afa0e7a"
+
```
Comment (silvin-lubecki) [inline Gopkg.toml]: nit: empty line

**Pull request merged**